includes/API/Operations/ApiLabkiOperationsStatus.php

```diff
@@ -0,0 +1,279 @@
+ * **Example Request:**
+ * ```
+ * GET api.php?action=labkiOperationStatus&limit=10
+ * ```
+ *
+ * **Response:**
+ * ```json
+ * {
+ *   "operations": [
+ *     { "operation_id": "...", "status": "success", ... },
+ *     { "operation_id": "...", "status": "running", ... }
+ *   ],
+ *   "count": 2,
+ *   "meta": {
+ *     "schemaVersion": 1,
+ *     "timestamp": "20251024122305"
+ *   }
+ * }
+ * ```
+ *
+ * ## Status Values
+ * - `queued`: Operation created but not yet started
+ * - `running`: Operation is in progress
+ * - `success`: Operation completed successfully
+ * - `failed`: Operation encountered an error
+ *
+ * ## Permissions
+ * - Users can view their own operations
+ * - Users with 'labkipackmanager-manage' permission can view all operations
+ *
+ * @ingroup API
+ */
+final class ApiLabkiOperationsStatus extends ApiBase {
+
+	public function __construct( ApiMain $main, string $name ) {
+		parent::__construct( $main, $name );
+	}
+
+	/** Execute API request. */
+	public function execute(): void {
+		$params = $this->extractRequestParams();
+		$operationId = isset( $params['operation_id'] ) ? trim( (string)$params['operation_id'] ) : null;
+		$limit = $params['limit'] ?? 50;
+		
+		// Validate limit range
+		if ( $limit < 1 || $limit > 500 ) {
+			$this->dieWithError( [ 'apierror-integer-outofrange', 'limit', 1, 500 ], 'badvalue' );
+		}
+
+		$registry = new LabkiOperationRegistry();
+		$currentUser = $this->getUser();
+		$currentUserId = $currentUser->getId();
+		$canManage = $currentUser->isAllowed( 'labkipackmanager-manage' );
+
+		// Single operation mode
+		if ( $operationId !== null && $operationId !== '' ) {
+			$this->executeSingleOperation( $registry, $operationId, $currentUserId, $canManage );
+			return;
+		}
+
+		// List operations mode
+		$this->executeListOperations( $registry, $currentUserId, $canManage, $limit );
+	}
+
+	/**
+	 * Handle single operation status query
+	 *
+	 * @param LabkiOperationRegistry $registry Operation registry instance
+	 * @param string $operationId Operation ID to query
+	 * @param int $currentUserId Current user's ID
+	 * @param bool $canManage Whether user has manage permission
+	 */
+	private function executeSingleOperation(
+		LabkiOperationRegistry $registry,
+		string $operationId,
+		int $currentUserId,
+		bool $canManage
+	): void {
+		$operation = $registry->getOperation( new OperationId( $operationId ) );
+
+		if ( $operation === null ) {
+			$this->dieWithError( 'labkipackmanager-error-operation-not-found', 'operation_not_found' );
+		}
+
+		// Permission check: users can only see their own operations unless they have manage permission
+		$operationUserId = $operation->userId() ?? 0;
+		// Deny access if: user is NOT a manager AND operation doesn't belong to current user AND operation is not system-owned
+		if ( !$canManage && $operationUserId !== $currentUserId && $operationUserId !== 0 ) {
+			$this->dieWithError( 'apierror-permissiondenied-generic', 'permission_denied' );
+		}
```

Comment:
permissions checks should come first - it should not be possible for a user without permissions to be able to tell if an item exists by getting a 404 if it doesn't and a 403 if it does. the reason why login forms say "either the email or password is invalid" rather than "the password is invalid" which confirms that an account with that email does exist